### PR TITLE
Added from dict and to_dict_extended

### PR DESCRIPTION
**Pull request by Khoroshevskyi**
I have added few functions. The most important are:
-  `to_dict_extended` returns extended dictionary of all variables in the object
-  `from_dict` initiates all variables in empty class from dictionary
-  `__eq__` function: checking equality of two objects:
There is a problem with this function. Some of the objects, even if they are the same, function returns False. The example of this bug : 
image
It says that this two values are not the same, but they are.

### CODE REVIEW

File: peppy/project.py [hunk 1]
```diff
@@ -141,18 +141,52 @@ def __init__(
...
+        dict_other["_samples"] = [s.to_dict() for s in other.samples]
+
+        return dict_self == dict_other
+
+    def to_dict_extended(self):
+        extended_dict = self.convert_to_dict(self)
+        extended_dict["_samples"] = [s.to_dict() for s in self.samples]
+        return extended_dict
+
+    def convert_to_dict(self, project_value=None):
```
[the remark this answers is not in the document]
Comment (nleroy917) [reply peppy/project.py:158]: I see there is also a `to_dict_extended` function. Maybe we combine them all and add some options to `to_dict`... Like maybe an `extended` flag? E.g.

```python
p = peppy.Project("/path/to/cfg")
p_dict = p.to_dict(extended=True)
```

**Review by nleroy917 — COMMENTED** [review]
I can try to test it a bit locally. We should create tests too

File: peppy/project.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Khoroshevskyi) [reply peppy/project.py:158]: Sounds good, I can change it. I have created to_dict  extended, as to_dict function wasn't returning all project values.
Also, I created 2 functions: 
`convert_to_dict` - converts all object variables to dict recursively.
`to_dict_extended` - mostly the same as `convert_to_dict`, but also adding corrected `_samples`.

So, should I change like that:  p.to_dict(extended=True)

**Comment by nleroy917** [comment]
### For note-taking purposes:
- The tests fail because our sample tables contain `nan` in them
- python will assume two `dict`s are equal if they have identical keys value pairs
- this fails when identical keys both have `nan`, since `nan` is by definition not equal to itself (see: [IEEE754](https://people.eecs.berkeley.edu/~wkahan/ieee754status/IEEE754.PDF))
- Thus, what **looks** like identical `dicts` are not identical in the eye's of python

Ways to potentially circumvent:
1. replace all `nan` values in sample tables with something like `0` or 'None`.
2. Have separate comparisons for sample tables and specifically look for `nan`

**Comment by nleroy917** [comment]
It looks like the issue of `nan` was addressed in #327. It seems to be causing a lot of problems. Is there a way to just recursively traverse the dictionary representation of a `PEP` and just convert all instances of `nan` to `None`, since `None == None >>> True`?

Is it possible to add a line [here](https://github.com/pepkit/peppy/blob/3326fe9ef306add344149064654dde84f5af0971/peppy/project.py#L189)... something like:

```python
if math.isnan(project_value):
    return None
```

**Pull request merged**